### ISSUE TITLE
HasOne and HasMany Relationships can only link to the Primary key of the child record.

### ISSUE THREAD

**Issue by IvandaNothabeer** [thedevdojo/voyager#2648]
- Laravel Version: 5.5.32
- Voyager Version: 1.0.15
- PHP Version: 7.0.23
- Database Driver & Version: MySQL 5.7.19

I have the following Parent - Child table structure ....

Parent
 - id, primary key, Auto-Increment
 - name, string
 - child_id, integer

Child
- id, primary key, Auto-Increment
- name, string
- parent_id, integer

The relationship is Parent HasOne/HasMany Child linked on Parent.child_id-->Child.parent_id

BREAD Relationships assume that the relationship is Parent.child_id -> Child.id

See relationship.blade.php, Line 48 ....

```
	@php 

		$relationshipData = (isset($data)) ? $data : $dataTypeContent;

		$model = app($options->model);

        	$query = $model::where($options->column, '=', $relationshipData->id)->first();

	@endphp
```

But this is not always true.

Need some method to select an alternative child id

**Comment by xembill** [comment]
Same issue for me. 
ı change manually
``
$query = $model::where($options->column, '=', $relationshipData->id)->first();
``
To
``
$query = $model::where($options->column, '=', $relationshipData->{$options->key})->first();
``

But this is not best solution.

**Comment by IvandaNothabeer** [comment]
I tested the same change and it does work :) , but really, this is just as inflexible as the current functionality.

IMO The Relationship configuration needs an additional parameter for the child table field_id.

**Comment by fletch3555** [comment]
This will likely not be fixed for 1.0.*

We're working on rebuilding the entire bread editor from scratch, so it'll function completely differently.  I haven't decided whether that'll find it's way into 1.1, 1.2, or 2.0 yet

**Comment by IvandaNothabeer** [comment]
For us, this is a critical function. We would like to add Voyager to several existing third party apps, but there's no possibility that the database structure can be changed.
